pkg/cluster/sync.go

```diff
@@ -469,11 +523,12 @@ func (c *Cluster) checkAndSetGlobalPostgreSQLConfiguration() error {
 		c.logger.Debugf("calling Patroni API on a pod %s to set the following Postgres options: %v",
 			podName, optionsToSet)
 		if err = c.patroni.SetPostgresParameters(&pod, optionsToSet); err == nil {
-			return nil
+			restartRequired = true
```

Comment:
Let's say that we changed `work_mem` from '4MB' to '16MB'.
Patroni will update `postgresql.conf` and `do pg_ctl reload`.
That doesn't require a restart.

Reply:
How we can distinguish parameters which require restart from those which don't?

Reply:
You have to wait for a while (ttl seconds should be enough), and call Patroni API on every pod:
```
$ curl -s http://localhost:8008/patroni | jq .
{
  "pending_restart": true,
```

If there is `pending_restart` flag is set - postgres in this pod must be restarted.

Also, the `/restart` [endpoint](https://patroni.readthedocs.io/en/latest/rest_api.html#restart-endpoint) expects to get the json object, where you can specify `{"restart_pending":true}`, so Patroni will restart postgres only if it is required.

Waiting for `ttl` seconds is mandatory.

Reply:
Added `{"restart_pending":true}` and waiting after restart.